cns/client_test.go

```diff
@@ -1414,3 +1482,78 @@ func isvSphereVersion70U3orAbove(ctx context.Context, aboutInfo vim25types.About
 	// For all other versions
 	return false
 }
+
+// isvSphereVersion80U3orAbove checks if specified version is 8.0 Update 3 or higher
+// The method takes aboutInfo{} as input which contains details about
+// VC version, build number and so on.
+// If the version is 8.0 Update 3 or higher, the method returns true, else returns false
+// along with appropriate errors during failure cases
+func isvSphereVersion80U3orAbove(ctx context.Context, aboutInfo vim25types.AboutInfo) bool {
+	items := strings.Split(aboutInfo.Version, ".")
+	version := strings.Join(items[:], "")
+	// Convert version string to string, Ex: "8.0.3" becomes 803, "8.0.3.1" becomes 703
+	if len(version) >= 3 {
+		vSphereVersionInt, err := strconv.Atoi(version[0:3])
+		if err != nil {
+			return false
+		}
+		// Check if the current vSphere version is 8.0.3 or higher
+		if vSphereVersionInt >= VSphere80u3VersionInt {
+			return true
+		}
+	}
+	// For all other versions
+	return false
+}
+
+// getServiceLocatorInstance returns a service locator instance for cross-VC operations.
+func getServiceLocatorInstance(u *url.URL, remoteVcClient *govmomi.Client) (*vim25types.ServiceLocator, error) {
```

Comment:
if you intend to use `getServiceLocatorInstance` outside of this repo, then it would be good to create a public method.

Reply:
Created a public method and moved it to `cns_util.go`.